### PR TITLE
src/utils.c: fix null reference error for g_propagate_prefixed_error

### PR DESCRIPTION
**Pull request by jungte**
It's my first PR. Please let me know if I did something wrong.

### CODE REVIEW

**Comment by ejoerns** [comment]
@jungte Thanks for the fix! Looks fine at first sight.

Please add your `Signed-off-by` line to the patch (`git commit --amend -s`): 

https://rauc.readthedocs.io/en/latest/contributing.html#developer-s-certificate-of-origin

You can simply force-push your branch to update the PR then.

**Comment by ejoerns** [comment]
The DCO check is not fully happy yet, since the mails do not match:

https://github.com/rauc/rauc/pull/999/checks?check_run_id=9133823272

Oh, still 1 PR to go...

**Comment by jluebbe** [comment]
You don't need to close the PR, you can just force-push to replace the commit.

**Comment by jungte** [comment]
OK, will try to do it later.

File: src/utils.c [hunk 1]
```diff
@@ -370,6 +370,11 @@ gboolean r_read_exact(const int fd, guint8 *data, size_t size, GError **error)
 					"Failed to read: %s", g_strerror(err));
 			return FALSE;
 		} else if (ret == 0) { /* end of file */
+			int err = errno;
```
Comment (jluebbe) [inline src/utils.c:373]: `errno` doesn't contain anything useful when pread returns 0, so we shouldn't try to add that to the error string.

File: src/utils.c [hunk 2]
```diff
@@ -370,6 +370,11 @@ gboolean r_read_exact(const int fd, guint8 *data, size_t size, GError **error)
...
+			g_set_error(error,
+					G_FILE_ERROR,
+					g_file_error_from_errno(err),
+					"End of file: %s", g_strerror(err));
```
Comment (jluebbe) [inline src/utils.c:377]: Perhaps
```suggestion
					"Unexpected end of file");
```

File: src/utils.c [hunk 3]
```diff
@@ -426,6 +431,11 @@ gboolean r_pread_exact(const int fd, guint8 *data, size_t size, off_t offset, GE
 					"Failed to read: %s", g_strerror(err));
 			return FALSE;
 		} else if (ret == 0) { /* end of file */
+			int err = errno;
```
Comment (jluebbe) [inline src/utils.c:434]: No `errno` here as well.

File: src/utils.c [hunk 4]
```diff
@@ -426,6 +431,11 @@ gboolean r_pread_exact(const int fd, guint8 *data, size_t size, off_t offset, GE
...
+			g_set_error(error,
+					G_FILE_ERROR,
+					g_file_error_from_errno(err),
+					"End of file: %s", g_strerror(err));
```
Comment (jluebbe) [inline src/utils.c:438]:
```suggestion
					"Unexpected end of file");
```

**Review by jluebbe — CHANGES_REQUESTED** [review]

**Comment by ejoerns** [comment]
@jungte would you like to incorporate the requested changes so we can merge this? Or should we?

**Comment by jungte** [comment]
@ejoerns Sure, I can do it in next few days if you can wait. The holiday is getting over.. :)

File: src/utils.c (hunk 1 above)
Comment (ejoerns) [reply src/utils.c:373]: @jungte this still remains as an unsued variable in your patch now. If you fix this also (and the occurrence noted below) than the PR should be ready for being merged.

Comment (jungte) [reply src/utils.c]: @ejoerns OK, how about this?

**Comment by ejoerns** [comment]
I've updated the branch to resolve a code style issue reported by uncrustify

**Review by jluebbe — APPROVED** [review]

**Pull request merged**